app/gui2/src/components/SvgIcon.vue

```diff
@@ -1,4 +1,10 @@
 <script setup lang="ts">
+/**
+ * A component displaying svg icon.
+ *
+ * It displays one group defined in `@/assets/icons.svg` file, specified by `variant` property.
+ *
+ */
 import icons from '@/assets/icons.svg'
 
 const props = defineProps<{ variant: string }>()
```

Comment:
nitpick: not sure "variant" is the right name to use here, to me it sounds like they're all slightly different variations of the same icon. i'd prefer `name` or `icon` instead (although they aren't that great either)

also might want to consider using a string union as the type instead. it'd be a decent amount of work to maintain, *and* you'd need to add a comment in the svg saying that this should be updated when a new icon is added...

... *but* i do think it'd make it a bit easier to use icons, partly because of the type errors but especially because of the autocomplete. that said, it might be too much work to maintain for too little benefit

Reply:
For me, the best name would be `Id` if it would not have other strong meaning in HTML. I will go back to `name`.

As for typed union: I must think about it. I'm just afraid it will prolong the PR too much, and gains are minuscule (The icon names should be copied from figma anyway). And perhaps the icons.svg needs some clean up anyway, I'm not sure, I must think about it.

Reply:
yeah, very fair. the typed union is definitely a minor improvement, plus we can always revisit it later if it does feel like it's needed